src/main/java/com/zontik/groshiky/dao/UserDao.java

```diff
@@ -18,4 +20,16 @@
     public void createUser(User user) {
         sessionFactory.getCurrentSession().save(user);
     }
-}
+
+    @Override
+    public User findByLogin(String login) {
+        try {
+            Session session = sessionFactory.getCurrentSession();
+            Query<User> findByUsername = session.createQuery("from User where login = :login");
+            findByUsername.setParameter("login", login);
+            return findByUsername.getSingleResult();
+        } catch (NoResultException ex) {
+            return null;
```

Comment:
1. We don't need NoResultException here (none of the methods above produces it)
2. It's not a good idea catch something and replac return value with null (in some cases it is possible, but better to avoid such solutions)
3. If you are using framework, try to find method you need (check if entity exists, get result without exception and ctr.) For example you can use uniqueResult() or uniqueResultOptional() methods (you can find it in Query)
4. You can choose to use uniqueResultOptional() method to be able to check if user exists and if so - get it (for future)

Reply:
fixt